src/EditorFeatures/Core.Wpf/Adornments/AdornmentManager.cs

```diff
@@ -22,42 +22,36 @@ namespace Microsoft.CodeAnalysis.Editor.Implementation.Adornments
     /// <summary>
     /// UI manager for graphic overlay tags. These tags will simply paint something related to the text.
     /// </summary>
-    internal class AdornmentManager<T> where T : GraphicsTag
+    internal abstract class AdornmentManager<T> where T : GraphicsTag
     {
         private readonly object _invalidatedSpansLock = new object();
 
         private readonly IThreadingContext _threadingContext;
 
-        /// <summary>View that created us.</summary>
-        private readonly IWpfTextView _textView;
-
-        /// <summary>Layer where we draw adornments.</summary>
-        private readonly IAdornmentLayer _adornmentLayer;
-
-        /// <summary>Aggregator that tells us where to draw.</summary>
-        private readonly ITagAggregator<T> _tagAggregator;
-
         /// <summary>Notification system about operations we do</summary>
         private readonly IAsynchronousOperationListener _asyncListener;
 
         /// <summary>Spans that are invalidated, and need to be removed from the layer..</summary>
         private List<IMappingSpan> _invalidatedSpans;
 
-        public static AdornmentManager<T> Create(
-            IThreadingContext threadingContext,
-            IWpfTextView textView,
-            IViewTagAggregatorFactoryService aggregatorService,
-            IAsynchronousOperationListener asyncListener,
-            string adornmentLayerName)
-        {
-            Contract.ThrowIfNull(threadingContext);
-            Contract.ThrowIfNull(textView);
-            Contract.ThrowIfNull(aggregatorService);
-            Contract.ThrowIfNull(adornmentLayerName);
-            Contract.ThrowIfNull(asyncListener);
+        /// <summary>View that created us.</summary>
+        protected IWpfTextView TextView { get; }
 
-            return new AdornmentManager<T>(threadingContext, textView, aggregatorService, asyncListener, adornmentLayerName);
-        }
+        /// <summary>Layer where we draw adornments.</summary>
+        protected IAdornmentLayer AdornmentLayer { get; }
+
+        /// <summary>Aggregator that tells us where to draw.</summary>
+        protected ITagAggregator<T> TagAggregator { get; }
```

Comment:
note (for the future): these could be protected readonly fields.  but props are also fine.